users/api/http/requests.go

```diff
@@ -17,6 +19,17 @@ type userReq struct {
 	user users.User
 }
 
+type viewUserInfo struct {
+	token string
+}
+
 func (req userReq) validate() error {
 	return req.user.Validate()
 }
+
+func (req viewUserInfo) validate() error {
+	if req.token == "" {
+		return users.ErrUnauthorizedAccess
+	}
+	return nil
+}
```

Comment:
Please keep struct and it's methods grouped together. Also, add Req at the end of the struct name.